Enable coupled forecast in sprint branch

WalterKolczynski-NOAA opened the pull request:
Updates the sprint branch to allow running the coupled model forecast (FV3-MOM6-CICE-WW3). Changes needed to run prototype 7c are below.

As part of this effort, resource settings for the forecast are consolidated into the config.resources file rather than being scattered across multiple configs. In order to handle resources correctly at workflow creation time when resources for all CDUMPs are needed, a list of cdumps is iterated over when CDUMP is empty. Resources for other ocn/ice resolutions than the prototype are just guesses and need to be optimized later.

Config files for components is now handled differently. The fcst config selectively sources component config files based on whether the corresponding component is active. This is a general solution of a control block that previously appeared in the j-job for the waves only. New files for ice and ocn are created and some settings have been moved there (though see Issue #429). Additional settings likely still need to be moved into these new files.

Instructions to run a prototype 7c analog:

1. Clone and run checkout, build_all, and link as normal for coupled:
(In sorc)
```
./checkout.sh
./build_all.sh -c
./link_workflow emc orion coupled
```

2. In ush/rocoto, run
```
./setup_expt_fcstonly.py --pslot $PSLOT --comrot $COMROT --expdir $EXPDIR --idate 2013040100 --edate 2013040100 --configdir ../../parm/config
```

3. In $EXPDIR/$PSLOT, make the following changes:
config.base
```
CASE=C384
FHMAX_GFS_??=48
FHOUT_GFS=6
FHOUT_HF_GFS=-1
APP=S2SW
```
config.fcst:
```
IALB=2
IEMS=2
DO_CA="YES"
```
config.fv3 for C384 case:
```
DELTIME=300
layout_x_gfs=12
layout_y_gfs=16
WRITE_GROUP_GFS=1
WRTTASK_PER_GROUP_GFS=88
WRTIOBUF="32M"
```

config.wave
```
waveGRD='gwes_30m'
waveesmfGRD=''
FHOUT_HF_WAV=3
export DTPNT_WAV=10800
OUTPARS_WAV="WND CUR ICE HS T01 T02 DIR FP DP PHS PTP PDIR CHA"
```
config.waveprep
```
IOSRV='1'
```

4. Back in ush/rocoto, run
```
./setup_workflow_fcstonly.py --expdir $EXPDIR/$PSLOT
```

5. In $EXPDIR/$PSLOT, modify $PSLOT.xml as follows:
- Remove or comment out the entire gfsinit task
- Remove or comment out gfsinit as a dependency for gfswaveprep
- Optionally reduce the walltime for the forecast job to 1 hour to move through the queue faster
-- The forecast job uses too many nodes for the debug queue, though you can use the debug queue for the wave prep jobs

6. In $ROTDIR (The first step only needs to be done once.):
```
cd $ROTDIR
ln -s /work/noaa/global/wkolczyn/noscrub/global-workflow/FV3ICS FV3ICS

mkdir -p $ROTDIR/$PSLOT/gfs.20130401/00/atmos
cd $ROTDIR/$PSLOT/gfs.20130401/00/atmos
ln -s $ROTDIR/FV3ICS/2013040100/atmos/C384/INPUT INPUT

mkdir -p $ROTDIR/$PSLOT/gdas.20130331/18/wave/restart
cd $ROTDIR/$PSLOT/gdas.20130331/18/wave/restart
ln -s $ROTDIR/FV3ICS/2013040100/wav/20130401.000000.restart.gwes_30m .
```

Closes #416

WalterKolczynski-NOAA committed:
Fix CCPP_SUITE definition

The specification of CCPP_SUITE in config.base was erroneously
using the CCPP_SUITES from the build system instead of specifying
a single CCPP_SUITE for the execution. This wasn't preventing the
atm-only from running, since the different variable name meant the
setting was just ignored.

Refs: #416

WalterKolczynski-NOAA committed:
Choose nems.configure template correctly

The confignamevarfornems was not being set correctly, so the default
of 'atm' was always used. In coupled-crow, this variable was set to
nems_temp in the j-job because there is a separate j-job for cold-
starting the mediator. Here I just cut out the middle-man and changed
the nems_temp assignments to confignamevarfornems in config.base.

Refs: #416

WalterKolczynski-NOAA committed:
Standardize partition settings

The partition settings in setup are updated so all slurm-based
machines are treated the same. Also fixes an issue where task-
specific partitions were created, but the task definition used
PARTITION_BATCH directly instead.

Refs: #417

WalterKolczynski-NOAA committed:
Fix error in workflow utils magic string

The command inserting the magic string no longer includes extraneous
characters. The workaround to catch key errors for those cases are
removed, so legitimate KeyErrors will now be properly raised.

Refs: #424

WalterKolczynski-NOAA committed:
Consolidate fcst resources and reorg configs

Resource settings for the forecast job were previously scattered
between multiple files. These settings are now consolidated in
config.resources.

The total number of PEs is now calculated properly if different
coupled models are included.

The config.fcst script is updated to properly source component
scripts if that component is on. New scripts for those components
were created where needed, and settings were moved into the
appropriate scripts (though it may be appropriate to relocate
additional settings in the future). The special load of the wave
config in the j-job is removed as it is no longer needed.

CDUMP assignment had to be moved up in the j-job to occur before
the config files are sourced so the correct settings are used.

Undefined DO switches for stochastic components are added to
prevent syntax errors downstream.

Settings for do_ugwp, do_tofd, and launch_level are removed from
the forecast config, as they are now set in the forecast script
based on the CCPP suite.

Refs: #416

WalterKolczynski-NOAA committed:
Turn off date cutoff for waves

Removing the bit that turns off the wave model prior to a certain
date, as the coupled model prototypes run far earlier than that
date. The code remains in commented form, as we may need to revisit
the cutoff and how to handle it going forward.

Refs: #416

WalterKolczynski-NOAA committed:
Fix forecast launcher for Orion

Updates the forecast launcher string on Orion so it uses the
correct number of tasks.

Refs: #416

WalterKolczynski-NOAA committed:
Fix WW3 input settings for coupled

Two settings specifying the type of external input to WW3 is updated
to use CPL when coupling with the associated model.

Refs: #416

WalterKolczynski-NOAA committed:
Change how indirect variable reference is used

Replaced the bash-specific indirect variable reference (${!<var>})
with one that uses eval (unfortunately). This was necessary because
the script is currently a korne-shell script at runtime, which does
not recognize the ! notation. Hopefully, this can be restored in the
future.

Refs: #416

WalterKolczynski-NOAA committed:
Properly define ocn/ice mesh filename

The ocn/ice mesh filename was not correctly being set.

Refs: #416

WalterKolczynski-NOAA committed:
Change default slow coupling interval

Changed the slow coupling interval (ocean coupling) from 60 min to
30 min to match value used in old coupled branch.

Refs: #416

WalterKolczynski-NOAA committed:
Coupled atm to wave for S2SW app

Turns on coupling from waves to atmosphere when running the full
coupled (S2SW) app. This includes moving the setting to config.base
so it is set alongside other app-specific settings.

Refs: #416

WalterKolczynski-NOAA committed:
Turn on lheatstrg and lseaspray

Changes the default setting for lheatstrg and lseaspray to match
those used in the latest prototype.

Refs: #416

WalterKolczynski-NOAA committed:
Fix mosaic and mesh files

The wrong path to the mosaic fix file was being used for coupled
forecasts and the mesh file was not being determined correctly based
on the ice resolution. Both are now corrected. Since it is determined
by the ice resolution, the ocn/ice meshfile name is now determined in
the CICE post-det.

Refs: #416

WalterKolczynski-NOAA committed:
Merge branch 'feature/coupled_sprint' into feature/sprint_coupled_fcst

WalterKolczynski-NOAA committed:
Remove unused variable in resources

Plan changed and _suffix was never used in the resources config,
so it is now removed.

Refs: #416

WalterKolczynski-NOAA committed:
Add trailing new line to ocean config

Refs: #416

WalterKolczynski-NOAA commented:
Forecast ran successfully on Orion until hitting the wallclock. Hoping forecast completes using a longer wallclock before Orion goes down for maintenance.

WalterKolczynski-NOAA committed:
Add coupled version of atm post script

A version of the atm post script is copied from coupled-crow branch.
It contains small differences from the normal script, and in the
future we might be able to consolidate back to one script.

Refs: #416

WalterKolczynski-NOAA commented:
Forecast completes, but the forecast script dies afterwards as it tries to copy the wrong ocean filename. Will revisit after Orion downtime.

Tries to copy `ocn_2013_04_01_05.nc` instead of `ocn_2013_04_01_03.nc` (the first ocean file).
```
2108 + for fhr in '$fhrlst'
2108 + export fhr=6
2108 + fhr=6
2108 + [[ 10#6 -ge 6 ]]
2108 + hh_inc_m=1
2108 + hh_inc_o=3
22108 + /apps/contrib/NCEP/libs/hpc-stack/intel-2018.4/prod_util/1.2.2/bin/ndate 6 2013040100
2108 + VDATE=2013040106
22108 + cut -c1-4
22108 + echo 2013040106
2108 + YYYY=2013
22108 + cut -c5-6
22108 + echo 2013040106
2108 + MM=04
22108 + echo 2013040106
22108 + cut -c7-8
2108 + DD=01
22108 + cut -c9-10
22108 + echo 2013040106
2108 + HH=06
2108 + SS=21600
22108 + /apps/contrib/NCEP/libs/hpc-stack/intel-2018.4/prod_util/1.2.2/bin/ndate -1 2013040106
2108 + m_date=2013040105
2108 + p_date=2013040106
22108 + cut -c1-4
22108 + echo 2013040105
2108 + year=2013
22108 + echo 2013040105
22108 + cut -c5-6
2108 + month=04
22108 + echo 2013040105
22108 + cut -c7-8
2108 + day=01
22108 + cut -c9-10
22108 + echo 2013040105
2108 + hh=05
2108 + export ocnfile=ocn_2013_04_01_05.nc
2108 + ocnfile=ocn_2013_04_01_05.nc
2108 + echo '/bin/cp -p -p ocn_2013_04_01_05.nc /work/noaa/global/wkolczyn/noscrub/global-workflow/add_coupled_2/gfs.20130401/00/ocn2013040106.01.2013040100.nc'
/bin/cp -p -p ocn_2013_04_01_05.nc /work/noaa/global/wkolczyn/noscrub/global-workflow/add_coupled_2/gfs.20130401/00/ocn2013040106.01.2013040100.nc
2108 + /bin/cp -p -p ocn_2013_04_01_05.nc /work/noaa/global/wkolczyn/noscrub/global-workflow/add_coupled_2/gfs.20130401/00/ocn2013040106.01.2013040100.nc
/bin/cp: cannot stat ‘ocn_2013_04_01_05.nc’: No such file or directory
```
```
Orion-login-1 fcst.12607/>ls ocn_2013*
ocn_2013_04_01_03.nc  ocn_2013_04_01_15.nc  ocn_2013_04_02_03.nc  ocn_2013_04_02_15.nc
ocn_2013_04_01_09.nc  ocn_2013_04_01_21.nc  ocn_2013_04_02_09.nc  ocn_2013_04_02_21.nc
```

JessicaMeixner-NOAA commented on jobs/JGLOBAL_FORECAST:
```diff
@@ -20,15 +26,6 @@ for config in $configs; do
     status=$?
     [[ $status -ne 0 ]] && exit $status
 done
-# Source additional configs
```
We need waves in the config for the forecast? Why was this removed?

WalterKolczynski-NOAA replied:
It was moved to config.fcst

JessicaMeixner-NOAA commented on parm/config/config.base.emc.dyn:
```diff
@@ -251,39 +252,40 @@ case "${APP}" in
-  export DO_WAVE="NO"
-  echo "WARNING: Wave suite turned off due to lack of RTOFS in GDA for SDATE"
-fi
+# if [[ "$SDATE" -lt "2019020100" ]]; then # no rtofs in GDA
```
This could be turned back on and an extra "if APP+ATMW" could be added.

JessicaMeixner-NOAA replied:
Will this be turned back on before merging by adding the If APP=ATMW conditional?

WalterKolczynski-NOAA replied:
TBD when we fix ATMW.

JessicaMeixner-NOAA commented on parm/config/config.fcst:
```diff
@@ -5,10 +5,26 @@
+esac
+
+# Source component configs if necessary
+for component in WAVE OCN ICE AERO; do
```
Why is this not done in the job scripts and instead hidden here?

WalterKolczynski-NOAA replied:
j-jobs are supposed to be as clean as possible, and there is the extra bit of stuff to determine if waves are on before this. Kate & I both agree this is just a better place for it.

aerorahul replied:
I agree w/ @JessicaMeixner-NOAA 
This should be part of the j-job where all the configs are sourced.
Adding this block is not really polluting the j-job.

WalterKolczynski-NOAA replied:
It's not just this block. There is a preceding block that changes DO_WAVE based on CDUMP and WAVE_CDUMP, so four more lines.

Also, config.fv3 is already sourced here first and pre-exists this change. We can move the whole lot to the j-job if that's what we really want to do, but moving here is consistent with what was here before. (config.fv3 should also be renamed config.atm to be consistent with the others.)

Given these, I would recommend keeping it as-is for now and then deciding whether we want to move the whole thing to the j-job for a future PR.

JessicaMeixner-NOAA replied:
So this is going to stay as is?  I still think this is now being hidden and becomes confusing for users.

WalterKolczynski-NOAA replied:
After discussing with Rahul, we decided it should stay as-is.

JessicaMeixner-NOAA replied:
I don't see any of that discussion here

WalterKolczynski-NOAA replied:
We discussed it on the g-w tag-up a couple weeks ago, but it was just repeating/elaborating on my points here.

JessicaMeixner-NOAA commented on parm/config/config.fv3:
```diff
@@ -47,8 +47,6 @@ case $case_in in
         export layout_y=2
         export layout_x_gfs=3
         export layout_y_gfs=2
-        export npe_wav=16
```
Where was this moved to instead?

WalterKolczynski-NOAA replied:
All PE stuff was moved to config.resources.

aerorahul replied:
Why is PE stuff there and Threading here?
Either both go there or both stay here.

WalterKolczynski-NOAA replied:
Only the wave PE count was moved. The atm PE count is derived in config.resources from the layout, threads, and writegroup settings that are currently still in config.fv3. There is no corresponding case block in config.resources, so one would have to be added to move nth_fv3 (or more) there. I'm also think it is a bit fuzzy as to what we should classify as a "resource" here vs a model parameter. The nth_fv3* goes, but what about the grid decomposition or the write groups?

This is another change we can tackle, but would recommend postponing it to a future PR.

aerorahul replied:
Ok. In that case please keep the current behavior. i.e. pes and threads in the same file.

WalterKolczynski-NOAA replied:
The number of wave PEs aren't (well, shouldn't be) related to the atm resolution and shouldn't have been in this file in the first place. That's the only thing that was moved out (and now it depends on the $waveGRD and not $CASE). So it wasn't just moving it, it was changing how wave PEs are calculated to something sensible.

JessicaMeixner-NOAA commented on parm/config/config.wave:
```diff
@@ -113,8 +113,17 @@ fi
 export WW3ATMINP='CPL'
-export WW3ICEINP='YES'
-export WW3CURINP='YES'
+if [[ $DO_ICE == "YES" ]]; then
```
This should be based on the app not on "DO_*" component variables.

WalterKolczynski-NOAA replied:
Would be interested in hearing why. While I'm not necessarily opposed, I also don't want that section to become overly large.

JessicaMeixner-NOAA replied:
What the wave model is coupled to or not is based on the APP, not on if "DO_COMP" variables existence.  This is also based on which nems.configure template is used and therefore would be more appropriate to be based on the apps.

WalterKolczynski-NOAA replied:
Open an issue and I'll change it. With the way the workflow is set up currently, it is functionally identical. Really it sounds like we should be determining the APP and everything from a nems.configure template name instead of the reverse.

WalterKolczynski-NOAA commented:
> Forecast completes, but the forecast script dies afterwards as it tries to copy the wrong ocean filename. Will revisit after Orion downtime.
> 
> Tries to copy `ocn_2013_04_01_05.nc` instead of `ocn_2013_04_01_03.nc` (the first ocean file).
> 
> ```
> 2108 + for fhr in '$fhrlst'
> 2108 + export fhr=6
> 2108 + fhr=6
> 2108 + [[ 10#6 -ge 6 ]]
> 2108 + hh_inc_m=1
> 2108 + hh_inc_o=3
> 22108 + /apps/contrib/NCEP/libs/hpc-stack/intel-2018.4/prod_util/1.2.2/bin/ndate 6 2013040100
> 2108 + VDATE=2013040106
> 22108 + cut -c1-4
> 22108 + echo 2013040106
> 2108 + YYYY=2013
> 22108 + cut -c5-6
> 22108 + echo 2013040106
> 2108 + MM=04
> 22108 + echo 2013040106
> 22108 + cut -c7-8
> 2108 + DD=01
> 22108 + cut -c9-10
> 22108 + echo 2013040106
> 2108 + HH=06
> 2108 + SS=21600
> 22108 + /apps/contrib/NCEP/libs/hpc-stack/intel-2018.4/prod_util/1.2.2/bin/ndate -1 2013040106
> 2108 + m_date=2013040105
> 2108 + p_date=2013040106
> 22108 + cut -c1-4
> 22108 + echo 2013040105
> 2108 + year=2013
> 22108 + echo 2013040105
> 22108 + cut -c5-6
> 2108 + month=04
> 22108 + echo 2013040105
> 22108 + cut -c7-8
> 2108 + day=01
> 22108 + cut -c9-10
> 22108 + echo 2013040105
> 2108 + hh=05
> 2108 + export ocnfile=ocn_2013_04_01_05.nc
> 2108 + ocnfile=ocn_2013_04_01_05.nc
> 2108 + echo '/bin/cp -p -p ocn_2013_04_01_05.nc /work/noaa/global/wkolczyn/noscrub/global-workflow/add_coupled_2/gfs.20130401/00/ocn2013040106.01.2013040100.nc'
> /bin/cp -p -p ocn_2013_04_01_05.nc /work/noaa/global/wkolczyn/noscrub/global-workflow/add_coupled_2/gfs.20130401/00/ocn2013040106.01.2013040100.nc
> 2108 + /bin/cp -p -p ocn_2013_04_01_05.nc /work/noaa/global/wkolczyn/noscrub/global-workflow/add_coupled_2/gfs.20130401/00/ocn2013040106.01.2013040100.nc
> /bin/cp: cannot stat ‘ocn_2013_04_01_05.nc’: No such file or directory
> ```
> 
> ```
> Orion-login-1 fcst.12607/>ls ocn_2013*
> ocn_2013_04_01_03.nc  ocn_2013_04_01_15.nc  ocn_2013_04_02_03.nc  ocn_2013_04_02_15.nc
> ocn_2013_04_01_09.nc  ocn_2013_04_01_21.nc  ocn_2013_04_02_09.nc  ocn_2013_04_02_21.nc
> ```

This issue seems to be related to a mismatch between FHOUT and FHRLST. I opened a new issue: #431

WalterKolczynski-NOAA committed:
Remove override of threads for C386 fcst

There was a line that was overriding the previous selection for the
number of threads when running at C384.

Also changed the settings to use the correct number of threads when
running the free forecast vs gdas.

Refs: #416

WalterKolczynski-NOAA committed:
Remove check for existence of tiled fix files

The existing of some fix files was being checked with fallback
values used if the files are missing. However, this approach does
not work with the tiled fix files, as they are passed as 'tileX',
while the files are tile1, tile2, etc. This was causing the files
checked to always fall back to their non-tiled versions.

In the future, we should check for the existence of all fix files,
not just the five that were checked here, appropriately handling
the tiled versions.

Refs: #416

KateFriedman-NOAA commented on ush/forecast_postdet.sh:
```diff
@@ -374,10 +374,6 @@ EOF
   FNSMCC=${FNSMCC:-"$FIX_AM/global_soilmgldas.statsgo.t${JCAP}.${LONB}.${LATB}.grb"}
 
   # If the appropriate resolution fix file is not present, use the highest resolution available (T1534)
-  [[ ! -f $FNALBC ]] && FNALBC="$FIX_AM/global_snowfree_albedo.bosu.t1534.3072.1536.rg.grb"
-  [[ ! -f $FNVETC ]] && FNVETC="$FIX_AM/global_vegtype.igbp.t1534.3072.1536.rg.grb"
-  [[ ! -f $FNSOTC ]] && FNSOTC="$FIX_AM/global_soiltype.statsgo.t1534.3072.1536.rg.grb"
-  [[ ! -f $FNABSC ]] && FNABSC="$FIX_AM/global_mxsnoalb.uariz.t1534.3072.1536.rg.grb"
```
Where are these set now?

WalterKolczynski-NOAA replied:
They're still set in the same place. These just changed them if the files didn't exist, but that doesn't work with the tiled files.

KateFriedman-NOAA replied:
Ah gotcha, thanks!

KateFriedman-NOAA commented on parm/config/config.ocn:
```diff
@@ -0,0 +1,10 @@
+#!/bin/ksh -x
+
+case "$CASE" in
```
Should the config.efcs block for OCNRES be merged in here?

WalterKolczynski-NOAA replied:
Possibly, but we can handle that later.

KateFriedman-NOAA commented on parm/config/config.fcst:
```diff
@@ -120,11 +90,7 @@ export h2o_phys=".true."
-export do_ugwp=".false."
-export do_tofd=".true."
-export launch_level=$(echo "$LEVS/2.35" |bc)
+# export launch_level=$(echo "$LEVS/2.35" |bc)
```
Where is this set now?

WalterKolczynski-NOAA replied:
These had already been changed to be set in parsing_namelist_FV3 at runtime based on gwd_opt (which is determined by the CCPP suite). This was dead code except the launch level

SMoorthi-emc commented:
Actually, the above statement is not accurate; if the tile files are not
available, the above hirres files would work.
sfcsub is more general.
Moorthi

On Wed, Aug 25, 2021 at 12:55 PM Walter Kolczynski - NOAA <
***@***.***> wrote:

> ***@***.**** commented on this pull request.
> ------------------------------
>
> In ush/forecast_postdet.sh
> <https://github.com/NOAA-EMC/global-workflow/pull/430#discussion_r695938731>
> :
>
> > -  [[ ! -f $FNALBC ]] && FNALBC="$FIX_AM/global_snowfree_albedo.bosu.t1534.3072.1536.rg.grb"
> -  [[ ! -f $FNVETC ]] && FNVETC="$FIX_AM/global_vegtype.igbp.t1534.3072.1536.rg.grb"
> -  [[ ! -f $FNSOTC ]] && FNSOTC="$FIX_AM/global_soiltype.statsgo.t1534.3072.1536.rg.grb"
> -  [[ ! -f $FNABSC ]] && FNABSC="$FIX_AM/global_mxsnoalb.uariz.t1534.3072.1536.rg.grb"
>
> They're still set in the same place. These just changed them if the files
> didn't exist, but that doesn't work with the tiled files.
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/NOAA-EMC/global-workflow/pull/430#discussion_r695938731>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ALLVRYQBX5XXLU5ZYYCTKLTT6UN7FANCNFSM5CWEXDYA>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&utm_campaign=notification-email>
> .
>


-- 
Dr. Shrinivas Moorthi
Research Meteorologist
Modeling and Data Assimilation Branch
Environmental Modeling Center / National Centers for Environmental
Prediction
5830 University Research Court - (W/NP23), College Park MD 20740 USA
Tel: (301)683-3718

e-mail: ***@***.***
Phone:  (301) 683-3718   Fax: (301) 683-3718

WalterKolczynski-NOAA commented:
> Actually, the above statement is not accurate; if the tile files are not available, the above hirres files would work. sfcsub is more general. Moorthi
> […](#)
> On Wed, Aug 25, 2021 at 12:55 PM Walter Kolczynski - NOAA < ***@***.***> wrote: ***@***.**** commented on this pull request. ------------------------------ In ush/forecast_postdet.sh <[#430 (comment)](https://github.com/NOAA-EMC/global-workflow/pull/430#discussion_r695938731)> : > - [[ ! -f $FNALBC ]] && FNALBC="$FIX_AM/global_snowfree_albedo.bosu.t1534.3072.1536.rg.grb" - [[ ! -f $FNVETC ]] && FNVETC="$FIX_AM/global_vegtype.igbp.t1534.3072.1536.rg.grb" - [[ ! -f $FNSOTC ]] && FNSOTC="$FIX_AM/global_soiltype.statsgo.t1534.3072.1536.rg.grb" - [[ ! -f $FNABSC ]] && FNABSC="$FIX_AM/global_mxsnoalb.uariz.t1534.3072.1536.rg.grb" They're still set in the same place. These just changed them if the files didn't exist, but that doesn't work with the tiled files. — You are receiving this because you are subscribed to this thread. Reply to this email directly, view it on GitHub <[#430 (comment)](https://github.com/NOAA-EMC/global-workflow/pull/430#discussion_r695938731)>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/ALLVRYQBX5XXLU5ZYYCTKLTT6UN7FANCNFSM5CWEXDYA> . Triage notifications on the go with GitHub Mobile for iOS <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675> or Android <https://play.google.com/store/apps/details?id=com.github.android&utm_campaign=notification-email> .
> -- Dr. Shrinivas Moorthi Research Meteorologist Modeling and Data Assimilation Branch Environmental Modeling Center / National Centers for Environmental Prediction 5830 University Research Court - (W/NP23), College Park MD 20740 USA Tel: (301)683-3718 e-mail: ***@***.*** Phone: (301) 683-3718 Fax: (301) 683-3718

The point is you can't just check for file names *tileX, because there is no *tileX; they are *tile1, *tile2, etc. I explained it more completely in the [commit message](https://github.com/NOAA-EMC/global-workflow/pull/430/commits/d83428c15203f60b6f2100293e8374af7428660f).

WalterKolczynski-NOAA committed:
Move lheatstrg with rest of lsm-dependent vars

The lheatstrg setting is added to the rest of the variables set
depending on the lsm type. Now overridden settings in config.fcst
are removed.

Refs: #416

WalterKolczynski-NOAA committed:
Modify ocean PETs to match prototypes

The numebr of PETs for the ocean model is changed to match the value used
by the prototypes.

Refs: #416

WalterKolczynski-NOAA committed:
Add runoff to data_table

Adds the runoff variable for the ocean model to data_table. (The table is
otherwise empty.)

Refs: #416

WalterKolczynski-NOAA committed:
Export stochastic switches from config.fcst

Exports are added to the DO switches for the stochastic physics
components so they actually make it into the forecast scripts.

Refs: #416

WalterKolczynski-NOAA commented:
I'm now able to bit-wise reproduce the couped-crow branch† on Orion:

DATA
- coupled-crow: `/work/noaa/stmp/wkolczyn/global-workflow/full_coupled/2013040100/gfs/fcst.125243`
- sprint_coupled_fcst: `/work/noaa/stmp/wkolczyn/RUNDIRS/add_coupled/2013040100/gfs/fcst.85503`

COM
- coupled-crow: `/work/noaa/global/wkolczyn/noscrub/global-workflow/full_coupled/gfs.20130401/00/`
- sprint_coupled_fcst: `/work/noaa/global/wkolczyn/noscrub/global-workflow/add_coupled/gfs.20130401/00`

EXPDIR
- coupled-crow: `/work/noaa/global/wkolczyn/save/global-workflow/expdir/full_coupled`
- sprint_coupled_fcst: `/work/noaa/global/wkolczyn/save/global-workflow/expdir/add_coupled`

Sprint is linking a bunch of unneeded files into the run directory (including a bunch of unneeded wave mod_defs), but they apparently aren't used since the output is identical. A couple of the namelist files have whitespace differences. Since the creation time is part of the ice history files, these were confirmed to be identical using ncdiff.

Downstream wave post jobs still aren't working.

```
diff global-workflow/full_coupled/2013040100/gfs/fcst.125243 RUNDIRS/add_coupled/2013040100/gfs/fcst.85503 -qr
Files global-workflow/full_coupled/2013040100/gfs/fcst.125243/ESMF_Profile.summary and RUNDIRS/add_coupled/2013040100/gfs/fcst.85503/ESMF_Profile.summary differ
Files global-workflow/full_coupled/2013040100/gfs/fcst.125243/fort.52 and RUNDIRS/add_coupled/2013040100/gfs/fcst.85503/fort.52 differ
Files global-workflow/full_coupled/2013040100/gfs/fcst.125243/history/iceh_06h.2013-04-01-21600.nc and RUNDIRS/add_coupled/2013040100/gfs/fcst.85503/history/iceh_06h.2013-04-01-21600.nc differ
Files global-workflow/full_coupled/2013040100/gfs/fcst.125243/history/iceh_06h.2013-04-01-43200.nc and RUNDIRS/add_coupled/2013040100/gfs/fcst.85503/history/iceh_06h.2013-04-01-43200.nc differ
Files global-workflow/full_coupled/2013040100/gfs/fcst.125243/history/iceh_06h.2013-04-01-64800.nc and RUNDIRS/add_coupled/2013040100/gfs/fcst.85503/history/iceh_06h.2013-04-01-64800.nc differ
Files global-workflow/full_coupled/2013040100/gfs/fcst.125243/history/iceh_06h.2013-04-02-00000.nc and RUNDIRS/add_coupled/2013040100/gfs/fcst.85503/history/iceh_06h.2013-04-02-00000.nc differ
Files global-workflow/full_coupled/2013040100/gfs/fcst.125243/history/iceh_06h.2013-04-02-21600.nc and RUNDIRS/add_coupled/2013040100/gfs/fcst.85503/history/iceh_06h.2013-04-02-21600.nc differ
Files global-workflow/full_coupled/2013040100/gfs/fcst.125243/history/iceh_06h.2013-04-02-43200.nc and RUNDIRS/add_coupled/2013040100/gfs/fcst.85503/history/iceh_06h.2013-04-02-43200.nc differ
Files global-workflow/full_coupled/2013040100/gfs/fcst.125243/history/iceh_06h.2013-04-02-64800.nc and RUNDIRS/add_coupled/2013040100/gfs/fcst.85503/history/iceh_06h.2013-04-02-64800.nc differ
Files global-workflow/full_coupled/2013040100/gfs/fcst.125243/history/iceh_06h.2013-04-03-00000.nc and RUNDIRS/add_coupled/2013040100/gfs/fcst.85503/history/iceh_06h.2013-04-03-00000.nc differ
Files global-workflow/full_coupled/2013040100/gfs/fcst.125243/history/iceh_ic.2013-04-01-00000.nc and RUNDIRS/add_coupled/2013040100/gfs/fcst.85503/history/iceh_ic.2013-04-01-00000.nc differ
Files global-workflow/full_coupled/2013040100/gfs/fcst.125243/ice_diag.d and RUNDIRS/add_coupled/2013040100/gfs/fcst.85503/ice_diag.d differ
Files global-workflow/full_coupled/2013040100/gfs/fcst.125243/ice_in and RUNDIRS/add_coupled/2013040100/gfs/fcst.85503/ice_in differ
Files global-workflow/full_coupled/2013040100/gfs/fcst.125243/input.nml and RUNDIRS/add_coupled/2013040100/gfs/fcst.85503/input.nml differ
Files global-workflow/full_coupled/2013040100/gfs/fcst.125243/logfile.000000.out and RUNDIRS/add_coupled/2013040100/gfs/fcst.85503/logfile.000000.out differ
Files global-workflow/full_coupled/2013040100/gfs/fcst.125243/log.gwes_30m and RUNDIRS/add_coupled/2013040100/gfs/fcst.85503/log.gwes_30m differ
Files global-workflow/full_coupled/2013040100/gfs/fcst.125243/log.mww3 and RUNDIRS/add_coupled/2013040100/gfs/fcst.85503/log.mww3 differ
Files global-workflow/full_coupled/2013040100/gfs/fcst.125243/mediator.log and RUNDIRS/add_coupled/2013040100/gfs/fcst.85503/mediator.log differ
Only in RUNDIRS/add_coupled/2013040100/gfs/fcst.85503: mod_def.aoc_9km
Only in RUNDIRS/add_coupled/2013040100/gfs/fcst.85503: mod_def.at_10m
Only in RUNDIRS/add_coupled/2013040100/gfs/fcst.85503: mod_def.ep_10m
Only in RUNDIRS/add_coupled/2013040100/gfs/fcst.85503: mod_def.glix_10m
Only in RUNDIRS/add_coupled/2013040100/gfs/fcst.85503: mod_def.glo_15mxt
Only in RUNDIRS/add_coupled/2013040100/gfs/fcst.85503: mod_def.glo_30m
Only in RUNDIRS/add_coupled/2013040100/gfs/fcst.85503: mod_def.gnh_10m
Only in RUNDIRS/add_coupled/2013040100/gfs/fcst.85503: mod_def.gsh_15m
Only in RUNDIRS/add_coupled/2013040100/gfs/fcst.85503: mod_def.wc_10m
Files global-workflow/full_coupled/2013040100/gfs/fcst.125243/nemsusage.xml and RUNDIRS/add_coupled/2013040100/gfs/fcst.85503/nemsusage.xml differ
Only in RUNDIRS/add_coupled/2013040100/gfs/fcst.85503: rmp_src_to_dst_conserv_002_001.nc
Only in RUNDIRS/add_coupled/2013040100/gfs/fcst.85503: rmp_src_to_dst_conserv_003_001.nc
Files global-workflow/full_coupled/2013040100/gfs/fcst.125243/ufs_model and RUNDIRS/add_coupled/2013040100/gfs/fcst.85503/ufs_model differ
Files global-workflow/full_coupled/2013040100/gfs/fcst.125243/ww3_multi.inp and RUNDIRS/add_coupled/2013040100/gfs/fcst.85503/ww3_multi.inp differ
```

† Coupled-crow as of two weeks ago, but with the post commit updated to match the sprint branch.

KateFriedman-NOAA reviewed (approved):
No further questions about this PR. Glad to hear results are bit-wise identical to f/c-c run!

WalterKolczynski-NOAA committed:
Fix default PEs when component not used

Changed the default number of PEs in nems.configure to be zero for
optional components. The old non-zero default causes problems when
waves are run without a ocean/ice model because the PE counts for
those components is still used for the PET bounds.

Refs: #416

WalterKolczynski-NOAA committed:
Fix coupling interval for atm-wav

The nems.configure template for atm-wave was still using the older
coupling_interval_sec, which was not being subsituted at runtime.
The parameter is updated to coupling_interval_slow_sec to match what
is used for the full coupled model, and the substitution is performed
whenever any coupling is on, instead of just cplflux.

Refs: #416

WalterKolczynski-NOAA committed:
Remove duplicate do_tofd from namelist

do_tofd was being added to the namelist twice when coupling was off.

Refs: #416

WalterKolczynski-NOAA committed:
Move OCNRES back to config.base

OCNRES is still needed in some instances even if the ocean model is
off, so the specification is moved back to config.base for now.

Refs: #416

WalterKolczynski-NOAA committed:
Merge branch 'feature/coupled_sprint' into feature/sprint_coupled_fcst

WalterKolczynski-NOAA commented:
Going to merge this PR while we continue to work on getting atm-wave figured out. It already isn't working in this branch (or develop), so this PR isn't breaking anything.
